### PR TITLE
Velero kotsadm restore item action

### PR DESCRIPTION
**Pull request by areed**
This add a [velero plugin](https://velero.io/docs/v1.6/overview-plugins/) to modify the kotsadm deployment at restore time. The plugin has been added to the kurl_util image, which is required to copy the binary into a specific directory in its default command in order to register the plugin when used as an initContainer in the velero deployment. The plugin looks up the modifications from a ConfigMap in the velero namespace that is generated when the kurl installer runs. The values that are modified on the restored kotsadm deployment are proxy env vars and the host CA path.

### CODE REVIEW

File: kurl_util/Makefile [hunk 1]
```diff
@@ -88,6 +88,9 @@ bin/network:
 bin/toml:
 	go build ${LDFLAGS} -o bin/toml cmd/toml/main.go
 
+bin/veleroplugin:
```
Comment (laverya) [inline kurl_util/Makefile:91]:
```suggestion
bin/veleroplugin: cmd/veleroplugin/main.go
```

File: kurl_util/cmd/veleroplugin/main.go [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+}
+
+func (p *restoreKotsadmPlugin) AppliesTo() (velero.ResourceSelector, error) {
+	return velero.ResourceSelector{
+		IncludedNamespaces: []string{"default"},
+		IncludedResources:  []string{"deployments"},
+	}, nil
+}
+
+func (p *restoreKotsadmPlugin) Execute(input *velero.RestoreItemActionExecuteInput) (*velero.RestoreItemActionExecuteOutput, error) {
```
Comment (laverya) [inline kurl_util/cmd/veleroplugin/main.go:47]: is there any way to make unit tests for functions like this?

**Review by laverya — APPROVED** [review]

File: kurl_util/Makefile (hunk 1 above)
Comment (areed) [reply kurl_util/Makefile]: added

File: kurl_util/cmd/veleroplugin/main.go (hunk 2 above)
Comment (areed) [reply kurl_util/cmd/veleroplugin/main.go:58]: added

**Pull request merged**